public/app/core/components/AccessControl/AddPermission.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserPicker } from 'app/core/components/Select/UserPicker';
+import { TeamPicker } from 'app/core/components/Select/TeamPicker';
+import { Button, Form, HorizontalGroup, Select } from '@grafana/ui';
+import { OrgRole } from 'app/types/acl';
+import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
+import { Assignments, PermissionTarget, SetPermission } from './types';
+
+const roles = [OrgRole.Admin, OrgRole.Editor, OrgRole.Viewer];
+
+export interface Props {
+  permissions: string[];
+  assignments: Assignments;
+  canListUsers: boolean;
+  onCancel: () => void;
+  onAdd: (state: SetPermission) => void;
+}
+
+export const AddPermission = ({ permissions, assignments, canListUsers, onAdd, onCancel }: Props) => {
+  const [target, setPermissionTarget] = useState<PermissionTarget>(PermissionTarget.User);
+  const [teamId, setTeamId] = useState<number>(0);
+  const [userId, setUserId] = useState<number>(0);
+  const [builtInRole, setBuiltinRole] = useState<string>('');
+  const [permission, setPermission] = useState<string>('');
+
+  const targetOptions = useMemo(() => {
+    const options = [] as any;
```

Comment:
```suggestion
    const options = [];
```

Not sure `any` is needed here

Reply:
I think i added that before adding to the array and the values could be inferred